cns/fsnotify/fsnotify.go

```diff
@@ -206,3 +224,34 @@ func (w *watcher) releaseIP(ctx context.Context, podInterfaceID, containerID str
 	}
 	return errors.Wrap(w.cli.ReleaseIPs(ctx, *ipconfigreq), "failed to release IP from CNS")
 }
+
+// call GetEndpoint API to get the state and then remove assiciated HNS
+func (w *watcher) deleteEndpoint(ctx context.Context, containerid string) error {
+	endpointResponse, err := w.cli.GetEndpoint(ctx, containerid)
+	if err != nil {
+		return errors.Wrap(err, "failed to read the endpoint from CNS state")
+	}
+	for _, ipInfo := range endpointResponse.EndpointInfo.IfnameToIPMap {
+		hnsEndpointID := ipInfo.HnsEndpointID
+		// we need to get the HNSENdpoint via the IP address if the HNSEndpointID is not present in the statefile
+		if ipInfo.HnsEndpointID == "" {
+			if hnsEndpointID, err = hnsclient.GetHNSEndpointbyIP(ipInfo.IPv4, ipInfo.IPv6, DefaultNetworkID); err != nil {
+				return errors.Wrap(err, "failed to find HNS endpoint with id")
+			}
+		}
+		w.log.Info("deleting HNS Endpoint with id ", zap.String("id", hnsEndpointID))
+		if err := hnsclient.DeleteHNSEndpointbyID(hnsEndpointID); err != nil {
+			return errors.Wrap(err, "failed to delete HNS endpoint with id "+ipInfo.HnsEndpointID)
+		}
+	}
+	return nil
+}
+
+// isStalessCNIMode verify if the CNI is running stateless mode
+func isStalessCNIMode(cnsconfig *configuration.CNSConfig) bool {
```

Comment:
```suggestion
func isStalessCNIWindows(cnsconfig *configuration.CNSConfig) bool {
```

Reply:
missed this one. Mad the change in new commit.